File: PING/PING.pl
#!/usr/bin/perl

# By Fred J. !!!

use Net::Ping;

autoflush STDOUT 1;

use constant false => 0;
use constant true  => 1;

$CO_V = "\033[1;32m"; #vert
$CO_R = "\033[1;31m"; #rouge
$CO_O = "\033[1;33m"; #orange
$CO_N = "\033[0;39m"; #normal

my $FILE_INIT   = 'host.ini';


# ---------------------------------- INIT ---------------------------
my $A = 1;

open (F, $FILE_INIT);
while ($LINE = <F>) {

        chomp $LINE;

        if (!grep(/^#/, $LINE) ) {

           $LINE =~ s/ //g;
           $LINE =~ s/\t//g;

           @LIST = split(';', $LINE);
           $IP   = $LIST[0];
           $DESC = $LIST[1] ."-". $LIST[2];

           if ($IP) {

                #print $A ." --> ". $IP ." - ". $DESC ." - ";
                printf ("%3s ->%-15s %-40s ", $A, $IP, $DESC);

                $TEST_ICMP = &ping($IP);

                #print "\t";
                print "\n";

                $A++;
           }
        } else {

#                print $LINE ."\n";

        }
}
close F;

print "\n";

exit;


# ------------------------------------------------------------------
sub ping {

        $IP = $_[0];


        $p = Net::Ping->new("icmp", 1);

        if ($p->ping($IP)) {
                print $CO_V ."OK". $CO_N;
                $TEST_ICMP = true;
        } else {
                print $CO_R ."HS". $CO_N;
                $TEST_ICMP = false;
        }

        return ($TEST_ICMP);

        $p->close();
        undef $p;
}

File: PING_TCP/ping.py
#!/usr/bin/python
# -*- coding:Utf-8 -*-

import socket
import os

my_conf = "host.ini"



CO_V = "\033[1;32m"; #vert
CO_R = "\033[1;31m"; #rouge
CO_O = "\033[1;33m"; #orange
CO_B = "\033[1;34m"; #bleu
CO_N = "\033[0;39m"; #normal


fo = open(my_conf, "r")
lines = fo.readlines()
fo.close


for i in range(	len(lines)):

	lines[i] = lines[i].rstrip()

	if ( lines[i][0] == '#' ):
		print CO_B + lines[i] + CO_N

	else:

		IP   = lines[i].split(';')[0]
		PORT = lines[i].split(';')[1]
		DESC = lines[i].split(';')[2]

		if ( PORT == 'icmp' ): 		#	Test Ping

			print IP +"\tICMP\t",
			
			response = os.system("ping -c 1 -W 1 " + IP +" >/dev/null")

			if response == 0:
   				print CO_V +" OK "+ CO_N,
			else:
   				print CO_R +" HS "+ CO_N,

			print "\t", DESC

		else:				#	Test TCP
			print IP +"\ttcp:"+ PORT +"\t",
	
			sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
			sock.settimeout(1)
			result = sock.connect_ex((IP , int(PORT) ))
			if result == 0:
   				print CO_V +" OPEN "+ CO_N,
			else:
   				print CO_R +" CLOSE"+ CO_N,

			print "\t", DESC

